sklearn/cluster/hierarchical.py

```diff
@@ -385,6 +385,11 @@ def linkage_tree(X, connectivity=None, n_components=None,
     --------
     ward_tree : hierarchical clustering with ward linkage
     """
+    if len(X.shape) != 2 or X.shape[0] != X.shape[1]:
+        raise ValueError('Matrix should be an upper triangular square matrix, as returned by pdist, '
+                         'found dimensionality %s' %
+                         'x'.join(map(str, X.shape)))
```

Comment:
This contradicts the docstring of the function and is actually breaking some tests.